gitAndTools.git-extras: 4.7.0 -> 5.0.0

cko opened the pull request:
###### Motivation for this change 
Update to latest version

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

teto commented:
changelog: https://github.com/tj/git-extras/releases/tag/5.0.0

teto commented:
https://github.com/tj/git-extras/blob/5.0.0/Installation.md doesn't list which as buildInput. they may need to fix it.

cko commented:
> changelog: https://github.com/tj/git-extras/releases/tag/5.0.0

I don't get what you want to say with this comment. Should something be added to this PR?

teto commented:
when reviewing, it helps knowing what changed to know where the potential problems are (espcially when there is a big version jump), putting it helps the reviewers. I think it's a good idea to put it in the commit message as well. I do look at git log and I am sure others do as well.

cko commented:
I rewrote the commit message.

teto commented:
@GrahamcOfBorg build gitAndTools.git-extras

teto commented:
there seems to be some issues

cko commented:
fixed :-)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review-may-2019/3032/79

c0bw3b commented on pkgs/applications/version-management/git-and-tools/git-extras/default.nix:
```diff
@@ -1,16 +1,22 @@
+  version = "5.0.0";
 
   src = fetchurl {
     url = "https://github.com/tj/git-extras/archive/${version}.tar.gz";
```
GitHub _archives_ are regenerated from time to time and the hash may change. You should rather use `fetchzip` or `fetchFromGitHub` here.
See #32997 for more context.

cko replied:
fixed

c0bw3b commented on pkgs/applications/version-management/git-and-tools/git-extras/default.nix:
```diff
@@ -1,16 +1,22 @@
+    patchShebangs .
+  '';
+
   installFlags = [ "DESTDIR=${placeholder "out"}" "PREFIX=" ];
```
From upstream install doc and a quick local test, it installs everything in the nix store as expected with just PREFIX flag
```nix
  installFlags = [ "PREFIX=${placeholder "out"}" ];
```

cko replied:
fixed

lilyball commented on pkgs/applications/version-management/git-and-tools/git-extras/default.nix:
```diff
@@ -1,16 +1,22 @@
   dontBuild = true;
 
+  preInstall = ''
+    patchShebangs .
```
Doesn't `patchShebangs` get run automatically? It seems to be part of the default `fixupOutputHooks` for everything using `stdenv.mkDerivation`.

cko replied:
without the explicit `patchShebangs` the package doesn't event build.

lilyball commented on pkgs/applications/version-management/git-and-tools/git-extras/default.nix:
```diff
@@ -1,16 +1,22 @@
+
   installFlags = [ "DESTDIR=${placeholder "out"}" "PREFIX=" ];
 
   postInstall = ''
```
This is only installing the zsh completion, but there's a bash one too. However it looks like `make install` installs the bash completion (but not the zsh one). Can we add a comment here explaining that the bash completion is already handled by `make install`?

cko committed:
gitAndTools.git-extras: 4.7.0 -> 5.0.0

For a list of changes and additions see: https://github.com/tj/git-extras/releases/tag/5.0.0
